src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/CovariantDerivOfExtrinsicCurvature.cpp

```diff
@@ -0,0 +1,136 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/CovariantDerivOfExtrinsicCurvature.hpp"
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace GeneralizedHarmonic {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ijj<DataType, SpatialDim, Frame> covariant_deriv_of_extrinsic_curvature(
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal_vector,
+    const tnsr::Ijj<DataType, SpatialDim, Frame>&
+        spatial_christoffel_second_kind,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept {
+  tnsr::ijj<DataType, SpatialDim, Frame> d_extrinsic_curvature{};
+  GeneralizedHarmonic::covariant_deriv_of_extrinsic_curvature<SpatialDim, Frame,
+                                                              DataType>(
+      make_not_null(&d_extrinsic_curvature), spacetime_unit_normal_vector,
+      spatial_christoffel_second_kind, extrinsic_curvature,
+      inverse_spacetime_metric, phi, d_pi, d_phi);
+  return d_extrinsic_curvature;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void covariant_deriv_of_extrinsic_curvature(
+    const gsl::not_null<tnsr::ijj<DataType, SpatialDim, Frame>*>
+        d_extrinsic_curvature,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal_vector,
+    const tnsr::Ijj<DataType, SpatialDim, Frame>&
+        spatial_christoffel_second_kind,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept {
+  if (UNLIKELY(get_size(get<0, 0, 0>(*d_extrinsic_curvature)) !=
```

Comment:
If you include `Utilities/ContainerHelpers.hpp` then you can do `destructive_resize_components(d_extrinsic_curvature, get<0>(spacetime_unit_normal_vector).size());` instead of the `if ...`

Reply:
done